lib/hal/fileio.c

```diff
@@ -89,6 +89,12 @@ static char *getCurrentDirString()
  */
 int XConvertDOSFilenameToXBOX(const char *dosFilename, char *xboxFilename)
 {
+	// Allow Xbox filenames directly
+	if (!memcmp(dosFilename, "\\Device\\", 8)) {
+		strcpy(xboxFilename, dosFilename);
+		return STATUS_SUCCESS;
+	}
```

Comment:
This is a pretty dirty hack to be honest: `\\Device\\` probably isn't a valid `DOSFilename` (which the function name implies).

I think it's fine, if we intend to remove `XConvertDOSFilenameToXBOX` in the future, or it might be fine if we wrap `XConvertDOSFilenameToXBOX` into a better named function.

Generally, this function is used for `fopen` etc. and I'm not sure if that should even accept the `\\Device\\` style filenames in the first place.

Reply:
@XboxDev thoughts?

Should we reject this altogether? If not, it should be merged.

Reply:
I'm also not sure what the use case is for passing in non DOS-style paths. Is there an issue that motivated this PR? Either way, I don't think it's a bad thing to be able to handle both styles.

Reply:
Current file IO API assumes DOS style paths and always calls this function leading to errors when Xbox style paths are supplied.

Reply:
Yes, but my question was: what are the cases when this function is called with Xbox-style paths?

Reply:
Only current case I know of is JayFoxRox/xbe-loader but it could probably be modified to use the NT functions directly if we don't like the changes from this PR. :+1: